transport/transport.go

```diff
@@ -63,13 +63,11 @@ type recvMsg struct {
 	err error
 }
 
-func (recvMsg) isItem() bool {
-	return true
-}
+func (*recvMsg) isItem() {}
 
 // All items in an out of a recvBuffer should be the same type.
 type item interface {
-	isItem() bool
```

Comment:
the method naming is probably confusing... I would expect isItem to return true, or false. I would not expect this method to exist as a marker item.